Incorrect param type

```
Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 908, in gevent._gevent_cgreenlet.Greenlet.run
  File "/opt/homebrew/Caskroom/miniforge/base/envs/moviebot/lib/python3.9/site-packages/socketio/server.py", line 731, in _handle_event_internal
    r = server._trigger_event(data[0], namespace, sid, *data[1:])
  File "/opt/homebrew/Caskroom/miniforge/base/envs/moviebot/lib/python3.9/site-packages/socketio/server.py", line 765, in _trigger_event
    return self.namespace_handlers[namespace].trigger_event(
  File "/opt/homebrew/Caskroom/miniforge/base/envs/moviebot/lib/python3.9/site-packages/flask_socketio/namespace.py", line 25, in trigger_event
    return self.socketio._handle_event(handler, event, self.namespace,
  File "/opt/homebrew/Caskroom/miniforge/base/envs/moviebot/lib/python3.9/site-packages/flask_socketio/__init__.py", line 828, in _handle_event
    ret = handler(*args)
  File "/Users/2920807/Repos/moviebot/moviebot/controller/server_socket.py", line 113, in on_message
    response = action(user_id, data["message"])
  File "/Users/2920807/Repos/moviebot/moviebot/controller/server_socket.py", line 173, in action
    return controller_flask.send_message(user_id, message)
  File "/Users/2920807/Repos/moviebot/moviebot/controller/controller_flask.py", line 229, in send_message
    self.continue_dialogue(user_id, payload)
  File "/Users/2920807/Repos/moviebot/moviebot/controller/controller_flask.py", line 150, in continue_dialogue
    ) = self.agent[user_id].continue_dialogue(
  File "/Users/2920807/Repos/moviebot/moviebot/agent/agent.py", line 221, in continue_dialogue
    agent_response, options = self.nlg.generate_output(
  File "/Users/2920807/Repos/moviebot/moviebot/nlg/nlg.py", line 326, in generate_output
    self._user_options_remove_preference(
  File "/Users/2920807/Repos/moviebot/moviebot/nlg/nlg.py", line 636, in _user_options_remove_preference
    param_key = DialogueAct(UserIntents.REMOVE_PREFERENCE, [param])
  File "/Users/2920807/Repos/moviebot/moviebot/dialogue_manager/dialogue_act.py", line 36, in __init__
    assert all(
AssertionError: All params should be of ItemConstraint type: ['genres']
```

Conversation:

```
User: hi
Agent: Do you have any specific genres in mind?
User: No
Agent: I could not find the genres No. Please answer a few questions to help me find a good movie. Can you give me a few keywords? For example, 'wrongful imprisonment' or 'crown jewels'.
User: bla bla
```

--- a/moviebot/nlg/nlg.py
+++ b/moviebot/nlg/nlg.py
@@ -633,7 +633,10 @@ class NLG:
                     negative = True
                     value = value.replace(".NOT.", "")
                 _a_an = "an" if value[0] in ["a", "e", "i", "o", "u"] else "a"
-                param_key = DialogueAct(UserIntents.REMOVE_PREFERENCE, [param])
+                param_key = DialogueAct(
+                    UserIntents.REMOVE_PREFERENCE,
+                    [ItemConstraint(param, Operator.EQ, value)],
+                )
                 if param == Slots.GENRES.value:
                     if negative:
                         options[param_key] = [

